README.md

```diff
@@ -92,14 +89,54 @@ rclnodejs.init().then(() => {
 });
 ```
 
-The benefits of using TypeScript become evident when working with more complex use-cases. The ROS 2 messages are defined in the `types/interfaces.d.ts` module. This module is updated as part of the `generate_messages` process. Here's a trivial example of working with a String msg.
+The benefits of using TypeScript become evident when working with more complex use-cases. ROS messages are defined in the `types/interfaces.d.ts` module. This module is updated as part of the `generate-ros-messages` process described in the next section. 
 
-```typescript
-const msg: rclnodejs.std_msgs.msg.String = {
-  data: 'hello ROS2 from rclnodejs',
-};
+## ROS2 Interface Message Generation (important)
+ROS components communicate by sending and receiving messages described 
+by the interface definition language (IDL). ROS client libraries such as 
+rclnodejs are responsible for converting these IDL message descriptions 
+into source code of their target language. For this, rclnodejs provides 
+the `generate-ros-messages` commandline script that reads in the IDL 
+messages files of a ROS environment and generates corresponding JavaScript 
+message interface files. Additionally, the tool generates the TypeScript 
+`interface.d.ts` file containing declarations for every IDL message file 
+processed. 
+
+Learn more about ROS interfaces and IDL [here](https://index.ros.org/doc/ros2/Concepts/About-ROS-Interfaces/).
+
+In the following example rclnodejs loads a generated JavaScript message file corresponding to the ROS `std_msgs/msg/String' definition.
+
+```
+import * as rclnodejs from 'rclnodejs';
+let stringMsgObject = rclnodejs.createMessageObject('std_msgs/msg/String');
+stringMsgObject.data = 'hello world';
 ```
 
+### Maintaining Generated Message JavaScript Files
+Message files are generated as a post-install step of the rclnodejs 
+installation process. Thereafter, you will need to manually run the 
+message generation script when new ROS message packages are installed 
+for which your ROS2-nodejs project has a dependency. 
+
+### Running `generate-ros-messages` Utility
+To use `generate-ros-messages` from your Nodejs package, create an npm 
+script entry in your package.json file as shown:
+
+```
+"scripts": {
+  "generate-ros-messages": "generate-ros-messages"
+  // your other scripts here
+}
+````
+
+To run the script use `npm` as follows:
+```
+npm run generate-ros-messages
+```
+The newly generated JavaScript files can be found at 
+`<yourproject>/node_modules/rclnodejs/generated/`.
+
+
```

Comment:
Could we split the current README into several docs and add the link into README because it's pretty long now :)

Reply:
Agree! It's got too much going on. How do envision restructuring it? Actually, that discussion is probably better deferred to a new issue on the topic.

Reply:
I'd like to keep the key information in the README and put other sections to other `.md`. How about moving the following sections into separate ones:

- Build from Scratch
- Troubleshooting
- Contributors (CONTRIBUTORS.md?)

I believe you are better than me in organizing the structures and make it more reasonable linguistically as a native speaker :smile:

Reply:
Agree. I'll start refactoring in the next 1 - 2 days. What for follow up questions about specific requirements or ideas that may come up in the process.

Reply:
Thanks, please take your time and enjoy the Christmas :christmas_tree: